### PR TITLE
feat(ckb-bin): add reset data subcommand

### LINKED ISSUE #1048: Add a subcommand to reset data
Add a new subcommand `reset-data`:

It **MUST** require users to confirm before deleting the data directory unless `--force` is given.

Usage:

```
ckb reset-data [--force] [--all | [--database | --indexer | --network | --network-peer-store | --network-secret-key | --logs ] ]

--force    delete data without interactive prompt
--all    delete the whole data directory
--database    delete both `data/db` and `data/indexer_db`
--indexer    delete only `data/indexer_db`
--network    delete both peer store and secret key
--network-peer-store    delete only `data/network/peer_store.db`
--network-secret-key    delete only `data/network/secret_key`
--logs    delete only `data/logs`
```

### PR DESCRIPTION
**Pull request by zjhmale**
Closes #1048

### CODE REVIEW

**Comment by doitian** [comment]
Please run `make fmt` and `make clippy` and fix the reported errors.

**Comment by zjhmale** [comment]
@doitian thx! I saw the CI logs.

**Comment by zjhmale** [comment]
termion does not support windows 😰

File: ckb-bin/src/subcommand/reset_data.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        prompt.push_str(&join_paths(target_files.clone()));
+        prompt.push_str("?");
+
+        write!(stdout, "{}", prompt).unwrap();
+        stdout.flush().unwrap();
+
+        let mut input = String::new();
+        let _ = stdin.read_line(&mut input);
+
+        if input != "y\n" {
```
Comment (doitian) [inline ckb-bin/src/subcommand/reset_data.rs]: I'm not sure whether it works in Windows, but `input.trim().to_uppercase() != "Y"` is a better choise.

Comment (zjhmale) [reply ckb-bin/src/subcommand/reset_data.rs]: oh, right, windows might be \r\n or something.

Comment (zjhmale) [reply ckb-bin/src/subcommand/reset_data.rs]: @doitian pls review again :)

File: ckb-bin/src/subcommand/reset_data.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    if args.logs {
+        if let Some(dir) = args.logs_dir {
+            target_dirs.push(dir);
+        }
+    }
+
+    if !args.force {
+        let mut prompt = String::from("remove ");
+        prompt.push_str(&join_paths(target_dirs.clone()));
```
Comment (quake) [inline ckb-bin/src/subcommand/reset_data.rs]: I don't think we need to clone `target_dirs` and `target_files` here, we can use `chain` and `to_string_lossy` to simplify the code:
```suggestion
    let to_be_deleted_targets_str = target_dirs.iter()
            .chain(target_files.iter())
            .map(|path| path.to_string_lossy())
            .collect::<Vec<_>>()
            .join(", ");
    
    let input = prompt(format!("remove {}?", to_be_deleted_targets_str));
```

Comment (zjhmale) [reply ckb-bin/src/subcommand/reset_data.rs]: sure

**Comment by doitian** [comment]
bors r=quake,doitian

**Pull request merged**